Assets/Scripts/PickupAmmunition.cs

```diff
@@ -0,0 +1,34 @@
+using UnityEngine;
+
+public class PickupAmmunition : MonoBehaviour
+{
+    public Transform hand;
+    public Vector3 inHandSize = new Vector3(0.2f, 0.2f, 0.2f);
+    public GameObject projectileHitParticles;
+
+    public Ammunition ammunition;
+
+    void Update()
+    {
+        if (!ammunition
+            && Input.GetButtonDown("Fire2")
+            && Physics.Raycast(Camera.main.ScreenPointToRay(Input.mousePosition), out var hitInfo, 2f, LayerMask.NameToLayer("Player")))
+        {
+            var collider = hitInfo.collider;
+            if (collider.CompareTag(Tags.AMMUNITION)
+                && !collider.GetComponent<Ammunition>())
+            {
+                var go = collider.gameObject;
+                ammunition = go.AddComponent<Ammunition>();
```

Comment:
Adding and removing components through code has led me personally through rabbitholes of debugging.
Suggest here to instead always have the Ammunition component on there, and call something like a "Pickup" method that does what the Start method does now.

Perhaps something like:

```cs
var ammunition = collider.gameObject.GetComponent<Ammunition>();
if (ammunition) {
    ammunition.Pickup(inHandSize);
}
```

Could probably discard the ammunition tag then as well